### PR TITLE
fix(vis): duplicate option values in scatterplot sidebar select

### PR DESCRIPTION
**Pull request by dv-usama-ansari**
Closes https://github.com/datavisyn/reprovisyn/issues/1686

### Developer Checklist (Definition of Done)

**Issue**

- [x] All acceptance criteria from the issue are met
- [x] Tested in latest Chrome/Firefox

**UI/UX/Vis**

- [x] Requires UI/UX/Vis review
  - [x] Reviewer(s) are notified @thinkh 
  - [ ] Review has occurred (_link to notes_)
  - [ ] Feedback is included in this PR
  - [ ] Reviewer(s) approve of concept and design

**Code**

- [x] Branch is up-to-date with the branch to be merged with, i.e., develop
- [x] Code is cleaned up and formatted
- [ ] Unit tests are written (frontend/backend if applicable)
- [ ] Integration tests are written (if applicable)

**PR**

- [x] Descriptive title for this pull request is provided (will be used for release notes later)
- [x] Reviewer and assignees are defined
- [x] Add type label (e.g., *bug*, *feature*) to this pull request
- [x] Add release label (e.g., `release: minor`) to this PR following [semver](https://semver.org/)
- [x] The PR is connected to the corresponding issue (via `Closes #...`)
- [x] [Summary of changes](#summary-of-changes) is written

### Summary of changes
Coming from: https://github.com/datavisyn/reprovisyn/issues/1686
Fixed issue where the user is unable to select multiple columns having the same name, but different descriptions

### Screenshots
See: https://github.com/datavisyn/reprovisyn/pull/1737

### Additional notes for the reviewer(s)
- This PR only affects the Scatter plot multiselect. Should there be similar issues in any other visualization, this fix can be applied and tested for the same :+1:

### CODE REVIEW

**Comment by dv-usama-ansari** [comment]
Found in https://github.com/datavisyn/reprovisyn/issues/1686:
Sometimes, the column info available for the visualization have the same `name`, but different `description` and `id`. In that case, the `MultiSelect` component in the sidebar behaves unexpectedly.
The problem due to this is, since the names are the same, the `MultiSelect` toggles the option instead of adding it as a another column.
In this PR, I have updated the `MulitSelect` to use `id` instead of name, since `id` is guaranteed to be different for each column. With this fix, the `MultiSelect` component behaves as expected.

File: src/vis/sidebar/MultiSelect.tsx [hunk 1]
```diff
@@ -26,9 +26,10 @@ export function MultiSelect({
     callback(currentSelected.filter((s) => s.id !== id));
   };
 
-  const handleValueSelect = (name: string) => {
-    const itemToAdd = filteredColumns.find((c) => c.info.name === name);
-    currentSelected.find((c) => c.name === name) ? handleValueRemove(itemToAdd.info.id) : callback([...currentSelected, itemToAdd.info]);
+  // NOTE: @dv-usama-ansari: See https://github.com/datavisyn/visyn_core/pull/261#issuecomment-2068786265
```
Comment (puehringer) [inline src/vis/sidebar/MultiSelect.tsx:29]: @dv-usama-ansari I think this note can be removed, as it doesn't provide too much value. I.e. there is no need to prove that it is better to use an id vs. the name :smile:

**Comment by dvmartinweigl** [comment]
this was also causing issues in bioinisght.
Tested your fix and it solves the issue :+1:

File: src/vis/sidebar/MultiSelect.tsx (hunk 1 above)
Comment (thinkh) [reply src/vis/sidebar/MultiSelect.tsx]: I removed the comment.

**Review by thinkh — APPROVED** [review]
Thanks for the fix. It works for me, too.

**Pull request merged**